homeassistant/components/airpatrol/climate.py

```diff
@@ -0,0 +1,321 @@
+        if humidity := climate_data.get("RoomHumidity"):
+            try:
+                return float(humidity)
+            except (ValueError, TypeError):
+                _LOGGER.error("Failed to convert humidity to float: %s", humidity)
+        return None
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        climate_data = self._unit.get("climate", {})
+        if temp := climate_data.get("RoomTemp"):
+            try:
+                return float(temp)
+            except (ValueError, TypeError):
+                _LOGGER.error("Failed to convert temperature to float: %s", temp)
+        return None
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the target temperature."""
+        climate_data = self._unit.get("climate", {})
+        params = climate_data.get("ParametersData", {})
+        if temp := params.get("PumpTemp"):
+            try:
+                return float(temp)
+            except (ValueError, TypeError):
+                _LOGGER.error("Failed to convert temperature to float: %s", temp)
+        return None
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return the current HVAC mode."""
+        climate_data = self._unit.get("climate", {})
+        params = climate_data.get("ParametersData", {})
+        pump_power = params.get("PumpPower", "off")
+        pump_mode = params.get("PumpMode", "heat")
+
+        if pump_power == "off":
+            return HVACMode.OFF
+        return HVAC_MODES.get(pump_mode, HVACMode.HEAT)
+
+    @property
+    def hvac_action(self) -> HVACAction | None:
+        """Return the current HVAC action."""
+        climate_data = self._unit.get("climate", {})
+        params = climate_data.get("ParametersData", {})
+        pump_power = params.get("PumpPower", "off")
+        pump_mode = params.get("PumpMode", "heat")
+
+        if pump_power == "off":
+            return HVACAction.OFF
+
+        if pump_mode == "heat":
+            return HVACAction.HEATING
+        if pump_mode == "cool":
+            return HVACAction.COOLING
+        return HVACAction.IDLE
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the current fan mode."""
+        climate_data = self._unit.get("climate", {})
+        params = climate_data.get("ParametersData", {})
+        fan_speed = params.get("FanSpeed", "max")
+        return FAN_MODES.get(fan_speed, FAN_HIGH)
+
+    @property
+    def swing_mode(self) -> str | None:
+        """Return the current swing mode."""
+        climate_data = self._unit.get("climate", {})
+        params = climate_data.get("ParametersData", {})
+        swing = params.get("Swing", "off")
+        return SWING_MODES.get(swing, SWING_OFF)
+
+    @property
+    def hvac_modes(self) -> list[HVACMode]:
+        """Return the list of available HVAC modes."""
+        return [HVACMode.HEAT, HVACMode.COOL, HVACMode.OFF]
```

Comment:
Set static properties with the shorthand attributes directly on the class instead ( `_attr_hvac_modes = ...` )